common/config/genericotlp.go

```diff
@@ -17,23 +23,63 @@ func (g *GenericOTLP) DestType() common.DestinationType {
 }
 
 func (g *GenericOTLP) ModifyConfig(dest ExporterConfigurer, currentConfig *Config) ([]string, error) {
-	url, exists := dest.GetConfig()[genericOtlpUrlKey]
+	config := dest.GetConfig()
+
+	url, exists := config[genericOtlpUrlKey]
 	if !exists {
 		return nil, errors.New("Generic OTLP gRPC endpoint not specified, gateway will not be configured for otlp")
 	}
 
-	grpcEndpoint, err := parseOtlpGrpcUrl(url, false)
+	tls := dest.GetConfig()[genericOtlpTlsKey]
+	tlsEnabled := tls == "true"
+
+	grpcEndpoint, err := parseOtlpGrpcUrl(url, tlsEnabled)
 	if err != nil {
 		return nil, errors.Join(err, errors.New("otlp endpoint invalid, gateway will not be configured for otlp"))
 	}
 
+	tlsConfig := GenericMap{
+		"insecure": !tlsEnabled,
+	}
+	caPem, caExists := config[genericOtlpCaPemKey]
+	if caExists && caPem != "" {
+		tlsConfig["ca_pem"] = caPem
+	}
+	insecureSkipVerify, skipExists := config[genericOtlpInsecureSkipVerify]
+	if skipExists && insecureSkipVerify != "" {
+		tlsConfig["insecure_skip_verify"] = insecureSkipVerify
+	}
```

Comment:
If `insecure_skip_verify` is expecting type boolean, then it might fail.
Consider using `parseBool` from utils here: https://github.com/odigos-io/odigos/blob/54b624935cb2398a8761b4c0bdc61f73dcc93506/common/config/utils.go#L129C6-L129C15

```suggestion
	if skipExists && insecureSkipVerify != "" {
		tlsConfig["insecure_skip_verify"] = parseBool(insecureSkipVerify)
	}
```

Reply:
From what i check the config value is of type `type GenericMap map[string]interface{}`, so it can hold both types. 
wdyt?